### PR TITLE
[SPARK-27561][SQL][FOLLOWUP] Move the two rules for Later column alias into one file

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Move the two rules `WrapLateralColumnAliasReference` and `ResolveLateralColumnAliasReference` into one file `ResolveLateralColumnAlias.scala`, instead of one rule in `Analyzer.scala` and the other one in another file.
Also update the code comments.

### Why are the changes needed?

Found this issue during reviewing https://github.com/apache/spark/pull/39040
This refactor should make the code easier to read and review.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @anchovYu  @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAlias.scala [hunk 1]
```diff
@@ -17,15 +17,17 @@
...
 import org.apache.spark.sql.catalyst.trees.TreeNodeTag
-import org.apache.spark.sql.catalyst.trees.TreePattern.LATERAL_COLUMN_ALIAS_REFERENCE
+import org.apache.spark.sql.catalyst.trees.TreePattern.{LATERAL_COLUMN_ALIAS_REFERENCE, OUTER_REFERENCE, UNRESOLVED_ATTRIBUTE}
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.apache.spark.sql.errors.QueryCompilationErrors
 import org.apache.spark.sql.internal.SQLConf
 
 /**
- * This rule is the second phase to resolve lateral column alias.
+ * The first phase to resolve lateral column alias.
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAlias.scala:30]: It is more clear to put the comment in the first phase rule

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
If this is a follow-up for SPARK-27561, the original PR (https://github.com/apache/spark/pull/39040) is not merged yet. Could you make a PR to him in order to commit as a single patch, @gengliangwang ?

**Comment by gengliangwang** [comment]
@dongjoon-hyun This one is follow-up of https://github.com/apache/spark/pull/38776. It doesn't contains the code changes of https://github.com/apache/spark/pull/39040

**Comment by dongjoon-hyun** [comment]
Then, let me revise the title of #39040 .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you for clarification, @gengliangwang .

**Comment by anchovYu** [comment]
This PR changes to use the resolver from the SimpleAnalyzer, which is always case sensitive one instead of the one that is determined by conf. This will cause lca unable to resolve case insensitive alias. I fix this issue in the new pr https://github.com/apache/spark/pull/39040/commits/136a9308e623fee5b1303103e6397f96d8bb6788 by refactoring the code.

**Comment by dongjoon-hyun** [comment]
Could you make a PR, @anchovYu ?

**Comment by cloud-fan** [comment]
Since this PR introduced a regression (case insensitive problem) and it's actually not necessary after my refactor https://github.com/apache/spark/pull/38888 , I'm reverting it.

**Comment by cloud-fan** [comment]
Reverted at https://github.com/apache/spark/commit/52082d3906bc3813cd3ff4447f7c75beb4f28612

**Pull request closed without merging**